actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -95,7 +95,11 @@ def tag_option(key, value, escape)
 
         private
           def prefix_tag_option(prefix, key, value, escape)
-            key = "#{prefix}-#{key.to_s.dasherize}"
+            if prefix == 'data' and key == :disable_with
+              key = "data-disable-with"
```

Comment:
See the full details in https://github.com/rails/rails/pull/30962/commits/a3f4e8a607bdef8870ed043f1981e03f957ae354 but essentially this provides a 4.41x speedup for strings that quality.